### PR TITLE
jax gradient

### PR DESCRIPTION
**Pull request by gshreya5**
- [ ] #13096

### CODE REVIEW

**Comment by Infrared1029** [comment]
Hi @gshreya5! thanks a lot for your contribution, can you make sure the tests are passing so I can take a closer look?

**Comment by gshreya5** [comment]
@Infrared1029 Hi can you help with the failing tests?

**Comment by Infrared1029** [comment]
hey @gshreya5, a quick look shows that you haven't used the actual `jax.numpy.gradient` signature which uses `varargs` instead of `spacing`, make sure you are copying the native function signature when creating it's frontend counterpart [check here](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.gradient.html). Also make sure you are passing the argument properly to the `ivy.gradient`, make sure to also specify the number of positional args in the test, if you are still stuck after these fixes let me know, but I'd suggest reading the deep dive section on frontends and their tests in the docs first.

**Comment by gshreya5** [comment]
@Infrared1029 Hi, modified as per your last comment,  still failing though,  can you check?

**Comment by Infrared1029** [comment]
Can you make sure you are setting the `num_positional_args` in the tests correctly? getting a proper test for `gradient` is a bit complicated, as `varargs` can take be defined in many ways, you can take on another function if this one is proving troublesome.
